File: sdk/typescript/src/resources/usage.ts
/**
 * Usage — spec §18 §2 canonical SDK resource.
 *
 * Two methods:
 *   current()           → org-level summary for current billing period
 *   forSession(sid)     → per-session breakdown
 */

import type { OmoiOSClient } from '../client.js';

export interface UsageSummary {
  organization_id?: string;
  subscription_tier?: string;
  workflows_used: number;
  workflows_limit: number;
  free_workflows_remaining: number;
  credit_balance: number;
  can_execute: boolean;
  reason: string;
}

export interface SessionUsage {
  session_id: string;
  compute_seconds: number;
  tokens_input: number;
  tokens_output: number;
  total_cost: number;
}

export class UsageResource {
  constructor(private readonly client: OmoiOSClient) {}

  /** Return the current billing-period summary for an org. */
  async current(organizationId?: string): Promise<UsageSummary> {
    const searchParams = organizationId
      ? { organization_id: organizationId }
      : undefined;
    const res = await this.client._request('GET', '/api/v1/usage', {
      searchParams,
    });
    return (await res.json()) as UsageSummary;
  }

  /** Return compute_seconds + token totals for a single session. */
  async forSession(sessionId: string): Promise<SessionUsage> {
    const res = await this.client._request(
      'GET',
      `/api/v1/usage/sessions/${sessionId}`,
    );
    return (await res.json()) as SessionUsage;
  }
}

File: sdk/python/omoios/resources/usage.py
"""Usage — spec §18 §2 canonical SDK resource.

Two methods:
  usage.current()           → org-level summary for current billing period
  usage.for_session(sid)    → per-session breakdown (compute_seconds,
                              tokens_input, tokens_output, total_cost)
"""

from __future__ import annotations

from typing import Any, Dict, Optional

from omoios.resources.base import BaseResource


class UsageResource(BaseResource):
    """Billing + per-session usage reads (spec §18 §2 `usage`)."""

    async def current(
        self, organization_id: Optional[str] = None
    ) -> Dict[str, Any]:
        """Return the current billing-period summary for an org.

        Omit `organization_id` to use the caller's first org membership.
        """
        params: Dict[str, Any] = {}
        if organization_id is not None:
            params["organization_id"] = organization_id
        response = await self._client._request(
            "GET", "/api/v1/usage", params=params or None
        )
        return response.json()

    async def for_session(self, session_id: str) -> Dict[str, Any]:
        """Return compute_seconds + token totals for a single session."""
        response = await self._client._request(
            "GET", f"/api/v1/usage/sessions/{session_id}"
        )
        return response.json()
